lib/elastic_record/as_document.rb

```diff
@@ -25,11 +25,12 @@ def elastic_search_value(field, mapping)
       value = try field
       return if value.nil?
 
-      value = case mapping[:type]&.to_sym
-              when :object
-                value.respond_to?(:as_search_document) ? value.as_search_document : value
-              when :nested
+      value = if mapping[:type]&.to_sym == :nested
                 value.map(&:as_search_document)
+              elsif value.respond_to?(:as_search_document)
+                value.as_search_document
+              elsif properties = mapping[:properties]&.keys
+                value.slice(*properties)
```

Comment:
That's a good question.

Reply:
All we can really say is that value *must* respond_to everything in `properties`.  If it doesn't, then the document was invalid to start.  We shouldn't assume it's a Hash though, only an object that doesn't respond to `as_search_document`.

All of this logic can be rewritten to reflect that.

Reply:
Nevermind!  I can't make this more clear.  I think assuming value is a hash with symbolized keys is still fair, though, because properties are defined as symbols.